cpp/include/milvus-storage/filesystem/s3/multi_part_upload_s3_fs.h

```diff
@@ -82,12 +76,17 @@ class S3ClientMetrics {
   std::atomic<int64_t> download_bytes{0};
 };
 
-class MultiPartUploadS3FS : public arrow::fs::S3FileSystem {
+class MultiPartUploadS3FS : public arrow::fs::FileSystem {
   public:
   ~MultiPartUploadS3FS() override;
 
   std::string type_name() const override { return MULTI_PART_UPLOAD_S3_FILESYSTEM_NAME; }
 
+  /// Return the original S3 options when constructing the filesystem
+  S3Options options() const;
```

Comment:
should be private function? 

also should return the rerference rathar then do the deep-copy?